components/calendar/src/week_of.rs

```diff
@@ -166,21 +166,24 @@ impl WeekCalculator {
     ///     },
     ///     week_calculator
     ///         .week_of_year(iso_date.day_of_year_info(), IsoWeekday::Friday)
-    ///         .unwrap()
     /// );
     /// ```
-    pub fn week_of_year(
-        &self,
-        day_of_year_info: DayOfYearInfo,
-        iso_weekday: IsoWeekday,
-    ) -> Result<WeekOf, CalendarError> {
+    pub fn week_of_year(&self, day_of_year_info: DayOfYearInfo, iso_weekday: IsoWeekday) -> WeekOf {
         week_of(
             self,
             day_of_year_info.days_in_prev_year,
             day_of_year_info.days_in_year,
             day_of_year_info.day_of_year,
             iso_weekday,
         )
+        .unwrap_or_else(|_| {
```

Comment:
Nit: use the specific error type like `RangeError` instead of `_`

Reply:
why?